modules/gentemplates/openapi3-doc/src/main/java/com/reprezen/genflow/openapi3/doc/ArrayHelper.xtend

```diff
@@ -17,13 +16,12 @@ class ArrayHelper implements Helper {
 	}
 
 	def String getArrayTypeSpec(Schema obj) {
-		val result = '''«obj.elementTypeName»«obj.arrayShape»'''
-		return result
+		'''«obj.elementTypeName»«obj.arrayShape»'''
 	}
 
 	def String getElementTypeName(Schema obj) {
 		val elementType = obj.elementType
-		#[elementType?.kaiZenSchemaName, elementType?.type, elementType.rzveTypeName].filter[it !== null].head
+		#[elementType?.type, elementType?.rzveTypeName].filterNull.last
```

Comment:
@ghillairet , if you can make your commit comments a bit more detailed, that would help. I'm trying to understand why `KaiZenParserHelper.getKaiZenSchemaName()` was causing a problem.  

AFAICT, that method gets the name of the property that contains the Schema Object as a value. Normally, for schemas contained in `/components/schemas`, this is a meaningful name. 

`AttributeHelper.getRzveTypeName()` will work, but only for schemas that have had their names set by `RepreZenVendorExtension.markTypeName()`. From the analysis I've done so far, it seems like names are only marked on inlined schemas. So I think `getKaiZenSchemaName` was included to cover non-inlined schemas.

I'm not 100% sure about my analysis at this point. You mentioned that things worked differently in OpenAPI2, so I'll take a look there and see if I can understand why it needs to change in OAS3.

Reply:
@tedepstein The OpenAPI3 doc generator was made before we had a normalizer for OpenAPI3. So getting type names from extensions added by the normalizer could not be done (this is what does the method `rzeTypeName`) as it is in v2 doc generator.